app/app/components/AddBookToBookshelf.jsx

```diff
@@ -0,0 +1,117 @@
+    const [searchResults, setSearchResults] = useState([]); // Store search results
+    const [loading, setLoading] = useState(false); // Loading state for API request
+    const [error, setError] = useState(null); // Error state
+    const [addedBooks, setAddedBooks] = useState([]); // Track added books
+
+    // Function to handle search input change
+    const handleSearchChange = (e) => {
+        setQuery(e.target.value);
+    };
+
+    // Function to add a book to the bookshelf based on category
+    const addBook = (book) => {
+        // Example logic for adding the book to the appropriate category
+        setAddedBooks((prev) => [...prev, { ...book, category }]); // Track added books by category
+        console.log(`Book added to ${category}:`, book);
+    };
+
+    // Fetch search results from the API when query changes
+    useEffect(() => {
+        if (query) {
+            const fetchSearchResults = async () => {
+                setLoading(true);
+                setError(null);
+                try {
+                    const response = await axios.get(
+                        "http://localhost:3001/api/searchGoogleBooks",
+                        {
+                            params: {
+                                query: query,
+                                page: 1,
+                                pageSize: 10,
+                            },
+                        }
+                    );
+                    setSearchResults(response.data || []);
+                } catch (err) {
+                    setError("An error occurred while fetching the search results.");
+                } finally {
+                    setLoading(false);
+                }
+            };
+            fetchSearchResults();
+        }
+    }, [query]);
+
+    return (
+        <div className={styles.addBookContainer}>
+            <h2>
+                Add Book to{" "}
+                {category === "read"
+                    ? "Read"
+                    : category === "currentlyReading"
+                      ? "Currently Reading"
+                      : "Wish to Read"}
+            </h2>
+            <input
+                type="text"
+                value={query}
+                onChange={handleSearchChange}
+                placeholder="Search for a book..."
+                className={styles.searchInput}
+            />
+            {loading && <p className={styles.loading}>Loading...</p>}
+            {error && <p className={styles.error}>{error}</p>}
+            {query && !loading && !error && (
+                <div className={styles.resultsContainer}>
+                    {searchResults.length > 0 ? (
+                        <ul className={styles.resultsList}>
+                            {searchResults.map((result, idx) => (
+                                <li key={idx} className={styles.resultItem}>
+                                    {result.cover_image && (
```

Comment:
For readability: This could be moved to a separate React component:
`SearchResultItem`